### ISSUE TITLE
Cannot stop(); event reactor not currently active

### ISSUE THREAD

**Issue by smarijic** [weiboad/kafka-php#194]
I am running async producer as a long running script. First and second batch of messages is published successfully, after that it fails with following LogicException

`"exception" => LogicException {
    #message: "Cannot stop(); event reactor not currently active"
    #code: 0
    #file: "vendor/amphp/amp/lib/NativeReactor.php"
    #line: 138
`
Using v0.2.0.8 of library.

**Comment by lcobucci** [comment]
@smarijic can you please send us a failing functional test that reproduces the behaviour your experiencing?

**Comment by smarijic** [comment]
I've tried to reproduce this behavior isolated from rest of the project and it turns out this exception is thrown when you are using async producer but set `$config->setIsAsyn(false)`, when this setting is not called, producer works normal so far. 
Your example for async producer shows this flag to be set to false.
I am not sure how to handle this. So far it looks like misconfiguration issue.

**Comment by lcobucci** [comment]
@smarijic humm tbqh I never managed to have something working with `$config->setIsAsyn(true)`... I'd suggest you to keep it set to `false` (even in the `README` it says to use it as `false` for async producer).

**Comment by smarijic** [comment]
Yes, my first implementation was based on README, but oddly for me it only works with isAsyn set to true. I'll create functional test where this issue in amphp occurs.

**Comment by lcobucci** [comment]
@smarijic awesome, thank you!
